Tests/CQRS.Tests/UserTests.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using AutoFixture.Xunit2;
+using Do_Svyazi.User.Application.CQRS.Users.Commands;
+using Do_Svyazi.User.DataAccess;
+using Do_Svyazi.User.Domain.Exceptions;
+using Do_Svyazi.User.Domain.Users;
+using EntityFrameworkCoreMock;
+using Assert = NUnit.Framework.Assert;
+
+namespace CQRS.Tests;
+
+public class UserTests
+{
+    private readonly Fixture _fixture;
+
+    public UserTests()
+    {
+        _fixture = new Fixture();
+        _fixture.Behaviors.OfType<ThrowingRecursionBehavior>().ToList()
+            .ForEach(b => _fixture.Behaviors.Remove(b));
+        _fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+    }
+
+    [Xunit.Theory, AutoData]
+    public async Task AddUser()
+    {
+        var dbContextMock = new DbContextMock<DoSvaziDbContext>();
+        dbContextMock.CreateDbSetMock(x => x.Users);
+
+        var addUserHandler = new AddUser.Handler(dbContextMock.Object);
+        var addUserCommand = new AddUser.Command("name1", "nickName1", "description1");
+        Guid userId = await addUserHandler.Handle(addUserCommand, CancellationToken.None);
+
+        var getUserHandler = new GetUser.Handler(dbContextMock.Object);
+        var getUserCommand = new GetUser.Command(userId);
+        MessengerUser user = await getUserHandler.Handle(getUserCommand, CancellationToken.None);
+
+        Assert.AreEqual(user.Id, userId);
+        Assert.AreEqual(user.Name, "name1");
+        Assert.AreEqual(user.NickName, "nickName1");
+        Assert.AreEqual(user.Description, "description1");
+    }
```

Comment:
```suggestion
    [Xunit.Theory, AutoData]
    public async Task AddUser(MessengerUser user)
    {
        var dbContextMock = new DbContextMock<DoSvaziDbContext>();
        dbContextMock.CreateDbSetMock(x => x.Users);

        var addUserHandler = new AddUser.Handler(dbContextMock.Object);
        var addUserCommand = new AddUser.Command(user.Name, user.NickName, user.Description);
        Guid userId = await addUserHandler.Handle(addUserCommand, CancellationToken.None);

        Assert.AreEqual(user.Name, user.Name);
        Assert.AreEqual(user.NickName, user.NickName);
        Assert.AreEqual(user.Description, user.Description);
    }
```

В рамках одного теста мы тестим только один функционал, создание юзера и получение - 2 разных функционала, разнеси это пж на разные тесты и старайся прокидывать параметры через конструктор (это то же самое, что если бы ты написал `fixture.Create<T>()`)

Reply:
P.S. `[Greedy]` - это штука, которая говорит fixture выбирать самый простой конструктор или что-то такое :)

Reply:
P.S.S. Я добавил пустой конструктор в юзера именно потому что fixture не мог создавать юзера со сложным конструктором (вроде как оно фиксится, но и так сойдёт)